shelley/chain-and-ledger/executable-spec/test/Rules/TestUtxow.hs

```diff
@@ -0,0 +1,167 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Rules.TestUtxow where
+
+import           Control.Monad (when)
+import           Data.Foldable (toList)
+import qualified Data.Map.Strict as Map (isSubmapOf)
+import qualified Data.Set as Set (intersection, isSubsetOf, map, null)
+
+import           Hedgehog (Property, forAll, property, withTests, (===))
+
+import           Control.State.Transition.Generator (trace)
+import           Control.State.Transition.Trace (TraceOrder (..), sourceSignalTargets, traceLength,
+                     traceSignals, _traceEnv)
+
+import           LedgerState (pattern UTxOState, decayedTx, keyRefunds)
+import           MockTypes (Tx, UTXOW)
+import           STS.Utxo (UtxoEnv (..))
+import           TxData (pattern TxIn, _body, _certs, _inputs, _txfee)
+import           UTxO (pattern UTxO, balance, deposits, txins, txouts)
+
+import           Ledger.Core (dom, (<|))
+
+------------------------------
+-- Constants for Properties --
+------------------------------
+
+numberOfTests :: Int
+numberOfTests = 300
+
+traceLen :: Int
+traceLen = 100
+
+--------------------------
+-- Properties for UTXOW --
+--------------------------
+
+-- | Preserve the balance in a transaction, i.e., the sum of the consumed value
+-- equals the sum of the created value.
+preserveBalance :: Property
+preserveBalance = withTests (fromIntegral numberOfTests) . property $ do
+  t <- forAll (trace @UTXOW $ fromIntegral traceLen)
```

Comment:
Why do we need `fromIntegral`? Can't we make `traceLen` have the same type `trace` expects?